Foundation/HTTPCookieStorage.swift

```diff
@@ -334,6 +334,28 @@ public extension Notification.Name {
 }
 
 extension HTTPCookie {
+    internal func validFor(host: String) -> Bool {
+        // RFC6265 - HTTP State Management Mechanism
+        // https://tools.ietf.org/html/rfc6265#section-5.1.3
+        //
+        // 5.1.3.  Domain Matching
+        // A string domain-matches a given domain string if at least one of the
+        // following conditions hold:
+        //
+        // 1)  The domain string and the string are identical.  (Note that both
+        //     the domain string and the string will have been canonicalized to
+        //     lower case at this point.)
+        //
+        // 2) All of the following conditions hold:
+        //    * The domain string is a suffix of the string.
+        //    * The last character of the string that is not included in the
+        //      domain string is a %x2E (".") character.
+        //    * The string is a host name (i.e., not an IP address).
+
+        let dotlessDomain = domain.first == "." ? String(domain.suffix(domain.count - 1)) : domain
```

Comment:
Perhaps `domain.hasPrefix(".") ? String(domain.dropFirst()) : domain`?

You may also wish to save the result of testing for the prefix since you do use it twice, or refactor to use it once (this can save you from allocating another `String` too):

```swift
guard domain.hasPrefix(".") else { return host == domain }
return host == domain.dropFirst() || host.hasSuffix(domain)
```

Reply:
That's a great suggestion. I have updated the code and will push shortly. I also updated unit tests a bit as part of this change.

Thanks!